ProteGO/Common/Validator/Validator.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import RxSwift
+
+final class Validator {
+    private(set) var type: ValidationType
+
+    init(type: ValidationType) {
+        self.type = type
+    }
+    func isValid(text: String?) -> Result<Bool, ValidationError> {
```

Comment:
Proponuję, żeby `isValid` przyjmowało `String` zamiast optionala.

Skoro wartość `.success` jest zawsze `true`, może warto zmienić wartość zwracaną na `Result<Void, ValidationError>`? W tym wypadku metoda mogłaby się nazwać `validate`.

Reply:
Co sądzicie o pomyśle użycia bardziej generycznego kodu: https://www.swiftbysundell.com/articles/using-errors-as-control-flow-in-swift/ , aby się nie blokować w dodawaniu warunków walidacji ?